pkg/cli/etcdsnapshot/etcd_snapshot.go

```diff
@@ -24,23 +43,18 @@ func Run(app *cli.Context) error {
 }
 
 func run(app *cli.Context, cfg *cmds.Server) error {
-	gspt.SetProcTitle(os.Args[0])
-
-	dataDir, err := server.ResolveDataDir(cfg.DataDir)
-	if err != nil {
+	if err := commandSetup(app); err != nil {
 		return err
 	}
 
-	nodeName := app.String("node-name")
-	if nodeName == "" {
-		h, err := os.Hostname()
-		if err != nil {
-			return err
-		}
-		nodeName = h
+	if len(app.Args()) > 0 {
+		return errors.New("the etcd-snapshot command does not take any arguments")
```

Comment:
Maybe make this an exported var somewhere so that we can reuse it for the other things that shouldn't be silently ignoring args?
```suggestion
		return errors.New("this command does not take any arguments")
```

Reply:
Added and will push in a moment.